### PR TITLE
[PGPNP] Support for AWS Secrets Manager

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

This PR adds support in the PGPNP Signer for [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) for retrieving from there the encryption private key. 

### Tested

Tested that the client can recover the secret. Log output of the test:
```
yarn run v1.22.4
$ yarn build && node dist/index.js
$ tsc -b .
84aa5096:: Initializing keystore
84aa5096:: Using AWS Secret Manager
84aa5096:: Fetching key
jcortejoso key: t0ps3cr3t
84aa5096:: Error retrieving key Error: Invalid private key
    at AWSKeyProvider.setPrivateKey (/celo-phone-number-privacy/signer/dist/key-management/key-provider-base.js:15:19)
    at AWSKeyProvider.fetchPrivateKeyFromStore (/celo-phone-number-privacy/signer/dist/key-management/aws-key-provider.js:29:18)
    at process._tickCallback (internal/process/next_tick.js:68:7)
84aa5096:: Fatal error occured. Exiting Error: CELO_PNP_ERR_11 Failed to retrieve key from keystore
    at AWSKeyProvider.fetchPrivateKeyFromStore (/celo-phone-number-privacy/signer/dist/key-management/aws-key-provider.js:33:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

(log output `jcortejoso key: ...` only added for testing, not included in pr)

### Pending changes

It is WIP a guide with some steps to deploy the signer in AWS (only documentation). From my point of view that guide can be added in another PR.

### CODE REVIEW

File: packages/phone-number-privacy/signer/src/key-management/aws-key-provider.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { SecretsManager } from 'aws-sdk'
+import { ErrorMessages } from '../common/error-utils'
+import logger from '../common/logger'
+import { KeyProviderBase } from './key-provider-base'
+
+export class AWSKeyProvider extends KeyProviderBase {
+  public async fetchPrivateKeyFromStore() {
+    try {
+      // Credentials are managed by AWS client as described in https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials-node.html
+      const region = process.env.KEYSTORE_AWS_REGION || 'us-east-1'
```
Comment (jmrossy) [inline packages/phone-number-privacy/signer/src/key-management/aws-key-provider.ts]: Please access these values from `config` like we do in the others and lets leave out the default values

File: packages/phone-number-privacy/signer/src/key-management/aws-key-provider.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        .getSecretValue({
+          SecretId: secretName,
+        })
+        .promise()
+
+      if (!privateKey.SecretString || !JSON.parse(privateKey.SecretString)[secretKey]) {
+        throw new Error('Key is empty or undefined')
+      }
+
+      this.setPrivateKey(JSON.parse(privateKey.SecretString)[secretKey])
```
Comment (jmrossy) [inline packages/phone-number-privacy/signer/src/key-management/aws-key-provider.ts]: Nit: Assign a var to the JSON.parse(privateKey.SecretString) so we don't have to do it twice.

**Review by jmrossy — COMMENTED** [review]
Looks great! Just two small comments :)

**Review by jmrossy — APPROVED** [review]
Ship it! 
You can fix the Install CI failure by running `yarn` from monorepo root to update the yarn.lock

**Review by aslawson — APPROVED** [review]
nice!

**Pull request merged**